### PR TITLE
refactor: PSG-5091  Remove JWK re-fetch logic

### PR DESCRIPTION
**Pull request by tdeshong**
BEGIN_COMMIT_OVERRIDE
feat: remove JWK re-fetch logic
END_COMMIT_OVERRIDE

## What's New?

Removes extra re-fetching call when JWK fails to find keyID on lookup. Refactor includes using `cacheSet` instead of individual cache and set.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have manually tested my code thoroughly
- [ ] I have added/updated inline documentation for public facing interfaces if relevant
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing integration and unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

## Additional context
Acceptance Criteria:
* [x] use JWT libraries to handle JWKS caching instead of doing it manually
* [x] Remove any refetch calls when the initial JWK call fails.
* [x] Return error.

### CODE REVIEW

**Review by ctran88 — COMMENTED** [review]
Would you be able to address the changes discussed in https://github.com/passageidentity/passage-go/pull/88#discussion_r1815404022 as well? Sorry, I forgot to mention this earlier

File: app.go [hunk 1]
```diff
@@ -42,14 +41,13 @@ func New(appID string, config *Config) (*App, error) {
...
 	}
 
-	app.jwksCache = jwk.NewCache(context.Background())
-	if err := app.jwksCache.Register(fmt.Sprintf(jwksUrl, appID)); err != nil {
+	cache := jwk.NewCache(context.Background())
+	if err := cache.Register(fmt.Sprintf(jwksUrl, appID)); err != nil {
 		return nil, err
 	}
 
-	if err := app.refreshJWKSCache(); err != nil {
```
Comment (ctran88) [inline app.go:50]: the cache needs to be manually [refreshed](https://github.com/passageidentity/passage-go/pull/95/files#diff-f2af39c59ae7258ce6dfa1f7fea0c60d8213467642c8bb936ca8339559524057L65-L67) just once before initializing the cached set

[reference](https://github.com/lestrrat-go/jwx/blob/536b643c661a886300246d2f334d4d836e61f497/docs/04-jwk.md?plain=1#L699-L705) from the example

File: app.go [hunk 2]
```diff
@@ -42,14 +41,13 @@ func New(appID string, config *Config) (*App, error) {
...
-		return nil, err
-	}
+	app.jwksCacheSet = jwk.NewCachedSet(cache, fmt.Sprintf(jwksUrl, appID))
+	var _ jwk.Set = app.jwksCacheSet
```
Comment (ctran88) [inline app.go]: this isn't necessary. it was only included in the [example](https://github.com/lestrrat-go/jwx/blob/536b643c661a886300246d2f334d4d836e61f497/docs/04-jwk.md?plain=1#L710) to show that it fulfills the `jwk.Set` interface

Comment (tdeshong) [reply app.go]: Removed this.

File: app.go (hunk 1 above)
Comment (tdeshong) [reply app.go:50]: Added back the refresh

**Review by ctran88 — APPROVED** [review]

**Pull request merged**